File: Selenium/Project/Project_Java/Activity9.java
package Project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity9 {
    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        // Navigate to the backend login page and log in
        driver.get("https://alchemy.hguy.co/jobs/wp-admin");
        WebElement usernameField = driver.findElement(By.id("user_login"));
        WebElement passwordField = driver.findElement(By.id("user_pass"));
        WebElement loginButton = driver.findElement(By.id("wp-submit"));

        usernameField.sendKeys("root");
        passwordField.sendKeys("pa$$w0rd");
        loginButton.click();

        // Locate the left-hand menu and click the menu item that says “Job Listings”
        WebElement jobListingsMenu = driver.findElement(By.xpath("//div[contains(@class, 'wp-menu-name') and text()='Job Listings']"));
        jobListingsMenu.click();

        // Locate the “Add New” button and click it
        WebElement addNewButton = driver.findElement(By.xpath("//a[@class='page-title-action']"));
        addNewButton.click();

        // Fill in the necessary details
        WebElement jobTitle = driver.findElement(By.id("job_title"));
        jobTitle.sendKeys("Backend Test Job");

        WebElement jobDescriptionFrame = driver.findElement(By.id("job_description_ifr"));
        driver.switchTo().frame(jobDescriptionFrame);
        WebElement jobDescription = driver.findElement(By.id("tinymce"));
        jobDescription.sendKeys("This is a backend test job description.");
        driver.switchTo().defaultContent();

        WebElement jobTypeDropdown = driver.findElement(By.id("job_type"));
        Select jobTypeSelect = new Select(jobTypeDropdown);
        jobTypeSelect.selectByVisibleText("Full Time");

        WebElement jobLocation = driver.findElement(By.id("job_location"));
        jobLocation.sendKeys("Backend Location");

        // Click the “Publish” button
        WebElement publishButton = driver.findElement(By.id("publish"));
        publishButton.click();

        // Verify that the job listing was created
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//div[@id='message']/p[contains(text(), 'Post published')]")));

        if (driver.getPageSource().contains("Backend Test Job")) {
            System.out.println("Job listing was created successfully.");
        } else {
            System.out.println("Job listing was not created.");
        }

        // Close the browser
        driver.quit();
    }
}

File: Selenium/Project/Python_Project/Activity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

# Initialize the WebDriver
driver = webdriver.Chrome()

try:
    # Navigate to the website's backend login page
    driver.get('https://alchemy.hguy.co/jobs/wp-admin/')
    
    # Find the username field of the login form and enter the username
    username_field = WebDriverWait(driver, 50).until(
        EC.presence_of_element_located((By.ID, "user_login"))
    )
    username_field.send_keys("root")  # Enter the username
    
    # Find the password field of the login form and enter the password
    password_field = driver.find_element(By.ID,"user_pass")
    password_field.send_keys("pa$$w0rd")  # Enter the password
    
    # Find the login button and click it
    login_button = driver.find_element(By.ID,"wp-submit")
    login_button.click()
    
    # Navigate to "Job Listings" in the left-hand menu
    job_listings_menu = WebDriverWait(driver, 10).until(
        EC.element_to_be_clickable((By.XPATH, "//div[contains(@class, 'wp-menu-name') and text()='Job Listings']"))
    )
    job_listings_menu.click()
    
    # Click the "Add New" button
    add_new_button = WebDriverWait(driver, 10).until(
        EC.element_to_be_clickable((By.CLASS_NAME, "page-title-action"))
    )
    add_new_button.click()
    
    # Fill in the necessary details
    title_field = WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.ID, "job_title"))
    )
    title_field.send_keys("Backend Test Job")
    
    description_field = driver.find_element(By.ID,"job_description_ifr")
    driver.switch_to.frame(description_field)
    description_body = driver.find_element("xpath","//body[@id='tinymce']")
    description_body.send_keys("This is a backend test job description.")
    driver.switch_to.default_content()
    
    # Click the "Publish" button
    publish_button = driver.find_element(By.ID,"publish")
    publish_button.click()
    
    # Verify that the job listing was created
    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.ID, "message")))
    assert "Job listing created" in driver.page_source, "Job listing creation failed"
    print("Job listing created successfully.")

finally:
    # Close the browser
    driver.quit()
